src/Twig/WidgetExtension.php

```diff
@@ -13,23 +13,40 @@
  */
 class WidgetExtension extends AbstractExtension
 {
+    /** @var Snippets */
+    private $snippets;
+
+    public function __construct(Snippets $snippets)
+    {
+        $this->snippets = $snippets;
+    }
     /**
      * {@inheritdoc}
      */
-    public function getFunctions()
+    public function getFunctions(): array
     {
         $safe = ['is_safe' => ['html']];
-        $env = ['needs_environment' => true];
 
         return [
-            new TwigFunction('countwidgets', [$this, 'dummy'], $safe + $env),
-            new TwigFunction('getwidgets', [$this, 'dummy'], $safe),
-            new TwigFunction('haswidgets', [$this, 'dummy'], $safe + $env),
-            new TwigFunction('widgets', [$this, 'dummy'], $safe + $env),
+            new TwigFunction('countwidgets', [$this, 'dummy'], $safe),
+            new TwigFunction('listwidgets', [$this, 'dummy'], $safe),
```

Comment:
this one does not need $safe

Reply:
neither do countwidgets and listwidgets..